plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8sauditeks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/source"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/symbols/extract"
+	"github.com/falcosecurity/plugins/plugins/k8saudit/pkg/k8saudit"
+	"github.com/falcosecurity/plugins/shared/go/aws/cloudwatchlogs"
+	"github.com/falcosecurity/plugins/shared/go/aws/session"
+	"github.com/invopop/jsonschema"
+)
+
+const pluginName = "k8saudit-eks"
+
+type Plugin struct {
+	k8saudit.Plugin
+	Logger *log.Logger
+	Config PluginConfig
+}
+
+type PluginConfig struct {
+	Profile         string `json:"profile"          jsonschema:"title=profile,description=The Profile to use to create the session, env var AWS_PROFILE if present"`
+	Region          string `json:"region"           jsonschema:"title=region,description=The Region of your EKS cluster, env var AWS_REGION is used if present"`
+	BufferSize      uint64 `json:"buffer_size"      jsonschema:"title=buffer_size,description=Buffer Size (default: 200),default=200"`
+	Shift           uint64 `json:"shift"            jsonschema:"title=shift,description=Time shift in past in seconds (default: 1s),default=1"`
+	PollingInterval uint64 `json:"polling_interval" jsonschema:"title=polling_interval,description=Polling Interval in seconds (default: 5s),default=5"`
+	UseAsync        bool   `json:"use_async"        jsonschema:"title=use_async,description=If true then async extraction optimization is enabled (default: true),default=true"`
+}
+
+func (k *Plugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:          9,
+		Name:        pluginName,
+		Description: "Read Kubernetes Audit Events for EKS from Cloudwatch Logs",
+		Contact:     "github.com/falcosecurity/plugins",
+		Version:     "0.1.0",
+		EventSource: "k8s_audit",
+	}
+}
+
+// Reset sets the configuration to its default values
+func (p *PluginConfig) Reset() {
+	if i := os.Getenv("AWS_DEFAULT_PROFILE"); i != "" {
+		p.Profile = i
+	}
+	if i := os.Getenv("AWS_PROFILE"); i != "" {
```

Comment:
if `AWS_PROFILE` is not presented, should we look up for other environment variable as well?

Reply:
what env vars are you thinking about?

Reply:
[AWS_WEB_IDENTITY_TOKEN_FILE](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html)

Reply:
technically, it will work, because I'm using the official SDK, if the env var is there, it will be used

Reply:
sgtm